### PR TITLE
Fix security issue

### PR DESCRIPTION
**Pull request by bartoszpietrzak1994**

### CODE REVIEW

File: features/being_unable_to_reorder_the_order_placed_by_another_customer.feature [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        And there is a customer "Rick Sanchez" identified by an email "rick.sanchez@wubba-lubba-dub-dub.com" and a password "Morty"
+        And there is a customer "Morty Smith" identified by an email "morty.smith@wubba-lubba-dub-dub.com" and a password "Rick"
+        And a customer "Morty Smith" placed an order "#00000666"
+        And the customer bought a single "Angel T-Shirt"
+        And the customer chose "Free" shipping method to "United States" with "Cash on Delivery" payment
+
+    @application
+    Scenario: Being unable to reorder the order placed by another customer
+        When the customer "rick.sanchez@wubba-lubba-dub-dub.com" tries to reorder the order "#00000666"
+        Then the order "#000001" should not be reordered
```
Comment (Zales0123) [inline features/being_unable_to_reorder_the_order_placed_by_another_customer.feature]: `#000001`? Shouldn't it be `#00000666`? 😄

File: spec/Checker/OrderCustomerRelationCheckerSpec.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+   function it_implements_order_customer_relation_checker_interface(): void
+   {
+       $this->shouldImplement(OrderCustomerRelationCheckerInterface::class);
+   }
+
+   function it_returns_true_when_order_was_placed_by_customer(
+       CustomerInterface $customer,
+       OrderInterface $order
+   ): void {
+        $customer->getId()->willReturn('1');
```
Comment (Zales0123) [inline spec/Checker/OrderCustomerRelationCheckerSpec.php]: Why string? `->willReturn(1);`

File: src/Checker/OrderCustomerRelationChecker.php [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+final class OrderCustomerRelationChecker implements OrderCustomerRelationCheckerInterface
+{
+    public function wasOrderPlacedByCustomer(OrderInterface $order, CustomerInterface $customer): bool
+    {
+        /** @var CustomerInterface $orderCustomer */
+        $orderCustomer = $order->getCustomer();
+
+        return
+            null != $customer &&
```
Comment (Zales0123) [inline src/Checker/OrderCustomerRelationChecker.php]: It cannot be null, as it's passed with function parameters

File: src/Checker/OrderCustomerRelationChecker.php [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+            null != $orderCustomer &&
```
Comment (Zales0123) [inline src/Checker/OrderCustomerRelationChecker.php]: Comparison should be strict `!==`

File: spec/Checker/OrderCustomerRelationCheckerSpec.php [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+        $order->getCustomer()->willReturn($customer);
+
+        $this->wasOrderPlacedByCustomer($order, $customer)->shouldReturn(true);
```
Comment (Zales0123) [inline spec/Checker/OrderCustomerRelationCheckerSpec.php]: I think something is wrong in this spec 😄 We should have 3 specifications here:

1. Order customer is the same as passed customer
```php
$orderCustomer()->getId()->willReturn(1);
$customer->getId()->willReturn(1);
```

2. Order customer is different than passed customer
```php
$orderCustomer()->getId()->willReturn(1);
$customer->getId()->willReturn(2);
```

3. Order has no customer assigned
```php
$order->getCustomer()->willReturn(null);
```

File: src/Resources/config/services.xml [hunk 6]
```diff
@@ -66,5 +68,7 @@
             <argument type="service" id="session"/>
         </service>
 
+        <service id="Sylius\CustomerReorderPlugin\Checker\OrderCustomerRelationChecker" />
+
```
Comment (Zales0123) [inline src/Resources/config/services.xml]: Redundant line imo

File: src/Checker/OrderCustomerRelationChecker.php (hunk 4 above)
Comment (bartoszpietrzak1994) [reply src/Checker/OrderCustomerRelationChecker.php]: You should discuss that with phpstan :D 

```
Strict comparison using !== between null and Sylius\Component\Core\Model\CustomerInterface will always evaluate to true.
```

$orderCustomer variable is typehinted a few lines above

**Pull request merged**